dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/task/AbstractJdbcTask.java

```diff
@@ -92,8 +101,10 @@ protected CSVFormat newCsvFormat(@Nonnull ResultSet rs) throws SQLException {
                     "If a custom query has been specified please confirm the selected columns match"
                         + " the following: ",
                     StringUtils.join(headerClass.getEnumConstants(), ", "))));
-    } else {
+    } else if (headerTransformer == null || headerTransformer.apply(rs) == null) {
       format = format.withHeader(rs);
```

Comment:
This is interesting: If we make the assumption that we're going to be error tolerant in headerTransformer, we should have a try/catch here, in which case it may be that the failure-case for headerTransformer is not just to use withHeader(rs) but something else? Because what caused the failure of headerTransformer might be rs.getColumnName(), which will just throw again in withHEader(rs).

So how about:
```
else if (headerTransformer != null) {
   try { // Do we actually want this try, or not?
       format = format.withHeader(headerTransformer.apply(rs));
    } catch (SQLException e) { /// something? Or should we just let this propagate }
}
else {
    format = format.withHeader(rs);
}

Reply:
I am voting for letting the `SQLException` just propagate especially because of the reason you outlined